internal/provider/resource/integration_aws_parameter_store.go

```diff
@@ -412,6 +410,26 @@ func (r *IntegrationAWSParameterStoreResource) Update(ctx context.Context, req r
 		}
 	}
 
+	updateIntegrationAuthRequest := infisical.UpdateIntegrationAuthRequest{
+		Integration:       infisical.IntegrationAuthTypeAwsSecretsManager,
+		IntegrationAuthId: plan.IntegrationAuthID.String(),
+	}
+	if authMethod == pkg.AwsAuthMethodAccessKey {
+		updateIntegrationAuthRequest.AccessId = plan.AccessKeyID.ValueString()
+		updateIntegrationAuthRequest.AccessToken = plan.SecretAccessKey.ValueString()
+	} else if authMethod == pkg.AwsAuthMethodAssumeRole {
+		updateIntegrationAuthRequest.AWSAssumeIamRoleArn = plan.AssumeRoleArn.ValueString()
+	}
+
+	_, err = r.client.UpdateIntegrationAuth(updateIntegrationAuthRequest)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating integration auth",
+			err.Error(),
+		)
+		return
+	}
```

Comment:
Better if we move this below the other validations so that we can somehow prevent inconsistency between TF state and integrationAuth state. Maybe we move it to directly before the ff:
<img width="1074" alt="image" src="https://github.com/user-attachments/assets/b0901c98-ce79-490a-9ead-da8dc6cd50ed">